### PR TITLE
Add conversion script from Jenkins config to Prow config

### PR DESCRIPTION
**Pull request by krzyzacy**
```
senlu@senlu:~/work/src/k8s.io/test-infra$ ./experiment/JenkinsToProw.py --jenkins=jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml --job=kubernetes-e2e-gce-flaky
- interval: 1h
  name: ci-kubernetes-e2e-gce-flaky
  spec:
    containers:
    - args:
      - --timeout=200
      - --bare
      env:
      - name: GOOGLE_APPLICATION_CREDENTIALS
        value: /etc/service-account/service-account.json
      - name: JENKINS_GCE_SSH_PRIVATE_KEY_FILE
        value: /etc/ssh-key-secret/ssh-private
      - name: JENKINS_GCE_SSH_PUBLIC_KEY_FILE
        value: /etc/ssh-key-secret/ssh-public
      image: gcr.io/k8s-testimages/kubekins-e2e-prow:0.3
      volumeMounts:
      - mountPath: /etc/service-account
        name: service
        readOnly: true
      - mountPath: /etc/ssh-key-secret
        name: ssh
        readOnly: true
      - mountPath: /root/.cache
        name: cache-ssd
    volumes:
    - name: service
      secret:
        secretName: service-account
    - name: ssh
      secret:
        defaultMode: 256
        secretName: ssh-key-secret
    - hostPath:
        path: /mnt/disks/ssd0
      name: cache-ssd

```

### CODE REVIEW

File: experiment/JenkinsToProw.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+# limitations under the License.
+
+"""Convert Jenkins config into prow config. """
+
+import argparse
+import copy
+import yaml
+
+
+TEMPLATE = {'name': '', 'interval': '1h', 'spec': {'containers': [{'image': 'gcr.io/k8s-testimages/kubekins-e2e-prow:0.3', 'args': [], 'volumeMounts': [{'readOnly': True, 'mountPath': '/etc/service-account', 'name': 'service'}, {'readOnly': True, 'mountPath': '/etc/ssh-key-secret', 'name': 'ssh'}, {'mountPath': '/root/.cache', 'name': 'cache-ssd'}], 'env': [{'name': 'GOOGLE_APPLICATION_CREDENTIALS', 'value': '/etc/service-account/service-account.json'}, {'name': 'JENKINS_GCE_SSH_PRIVATE_KEY_FILE', 'value': '/etc/ssh-key-secret/ssh-private'}, {'name': 'JENKINS_GCE_SSH_PUBLIC_KEY_FILE', 'value': '/etc/ssh-key-secret/ssh-public'}]}], 'volumes': [{'secret': {'secretName': 'service-account'}, 'name': 'service'}, {'secret': {'defaultMode': 256, 'secretName': 'ssh-key-secret'}, 'name': 'ssh'}, {'hostPath': {'path': '/mnt/disks/ssd0'}, 'name': 'cache-ssd'}]}}
```
Comment (spxtr) [inline experiment/JenkinsToProw.py]: Can you format this more reasonably?

Comment (krzyzacy) [inline experiment/JenkinsToProw.py]: awwwwww that looks more terrible on github

Comment (krzyzacy) [inline experiment/JenkinsToProw.py]: beautified

**Comment by krzyzacy** [comment]
/assign @spxtr @fejta 

this script should be usable.

```
./experiment/JenkinsToProw.py --config=jobs/config.json --delete --jenkins=jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml --prow=prow/config.yaml --job=kubernetes-e2e-gce-alpha-features-release-1.4
```

for the converted job I'm getting
```
ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain
```
am I still missing some project settings?

**Comment by fejta** [comment]
`JenkinsToProw.py` can you change this to `jenkins_to_prow.py`

**Comment by krzyzacy** [comment]
> JenkinsToProw.py can you change this to jenkins_to_prow.py

done

**Comment by fejta** [comment]
/lgtm

Anything interesting to unit test?

**Comment by krzyzacy** [comment]
> Anything interesting to unit test?

Let me try a couple more jobs, then add the test in a separate PR.

**Pull request merged**